scripts/data/online_evals_periodic_eval_chron.py

```diff
@@ -0,0 +1,131 @@
+from datetime import datetime
+import os
+from getpass import getpass
+
+import pandas as pd
+import phoenix as px
+
+from phoenix.session.evaluation import get_qa_with_reference, get_retrieved_documents
+from phoenix.trace import DocumentEvaluations, SpanEvaluations
+from phoenix.experimental.evals import (
+    HallucinationEvaluator,
+    OpenAIModel,
+    QAEvaluator,
+    RelevanceEvaluator,
+    run_evals,
+)
+from phoenix.trace import TraceDataset
+from datetime import datetime, timedelta
+
+
+#Optional but speeds up Evals by 10x
+import nest_asyncio
+nest_asyncio.apply()
+
+def lookup_traces(session):
+    # Get traces into a dataframe
+    #spans_df = session.get_spans_dataframe("span_kind == 'RETRIEVER'")
+    spans_df = session.get_spans_dataframe() #all spans
+    trace_df = session.get_trace_dataset()
+    if not trace_df:
+        return None, None
+    evals = trace_df.evaluations
+    evaluation_dfs = []
+    for eval in evals:
+        eval_dict = eval.__dict__
+        eval_df = eval_dict["dataframe"]
+        # all dataframes have a tuple index where index[0] is uuid, we'll use this to look for them in spans_df
+        evaluation_dfs.append(eval_df)
+
+    if spans_df is None:
+        return None
+    spans_df['date'] = pd.to_datetime(spans_df['end_time']).dt.date
+
+    # Get today's date
+    today_date = datetime.now().date() + timedelta(days=1)
+    # Calculate yesterday's date
+    yesterday_date = today_date - timedelta(days=1)
+    # Filter for entries from the last day (i.e., yesterday and today)
+    selected_date_spans_df = spans_df[(spans_df['date'] == today_date) | (spans_df['date'] == yesterday_date)]
+    return selected_date_spans_df, evaluation_dfs
+
+
+if __name__ == "__main__":
+    if os.environ.get("OPENAI_API_KEY") is None:
+        openai_api_key = getpass("🔑 Enter your OpenAI API key: ")
+        os.environ["OPENAI_API_KEY"] = openai_api_key
```

Comment:
This doesn't make a ton of sense - these are scripts. Shouldn't this just be a pre-requisite and you throw?